### PR TITLE
History

### PR DESCRIPTION
**Pull request by mansipadgelwar**
History Management
1. When I view a particular video I will add it to the history.
2. I can delete a particular video from the history
3. I can see a button to "Clear full history" which will clear all the history at once.

### CODE REVIEW

**Review by ajay-mandaviya — APPROVED** [review]

**Review by ajay-mandaviya — COMMENTED** [review]
How to close playlist modal  in playlist page 
just at  look this page

**Comment by mansipadgelwar** [comment]
> How to close playlist modal in playlist page just at look this page

fixed

**Review by MonikaJhingan — APPROVED** [review]
Code is well structured ui looks good needs to be merged.

**Review by dev-enforced — APPROVED** [review]
The code structure looks really good and well structured. Liked the way you have made a custom loader to show the video list in the different sections but couldn't get to see it due to data coming up fast from the server. Rest the history feature is also working good!!
Midori Library Live preview : https://deploy-preview-6--midori-video-library.netlify.app/ is also working . Can add a favicon too with title changed to Midori Videos
Do add a _redirects file in public folder to avoid Page Not found errors from Netlify end. 
The playlist modal has also got fixed thanks to Ajay's feedback and is working effectively.
And create Playlists is also working correctly.
Some things which I noticed which can be worked on the later stages:
1) Once we go to a particular video's page view that video gets added to the history. Until the video is not being played because the player is initially at pause is there a way to avoid adding that page to the history
2) Another thing is that when I visit an already added to history section video it shows a toast that unable to add this video to history indicating some  error while actually there isn't so instead you can show up a toast that the video is already there in history in light blue color.

Awesome work @mansipadgelwar learnt few important things related to loader implementation while data is being fetched.

**Review by MonikaJhingan — APPROVED** [review]
Code is well structured ui looks good.merge it.

File: src/components/PlaylistForm/PlaylistForm.jsx [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import { useState } from "react";
 import "../PlaylistForm/PlaylistForm.css";
-import axios from "axios";
 import { useAuth } from "../../context/authContext/authenticationContext";
 import { useServices } from "../../context/servicesContext/servicesContext";
```
Comment (harshalini) [inline src/components/PlaylistForm/PlaylistForm.jsx:4]: You can import useAuth, useServices in one file and then export from there

File: src/components/VideoCard/VideoCard.jsx [hunk 2]
```diff
@@ -1,7 +1,32 @@
 import "../VideoCard/VideoCard.css";
 import { Link } from "react-router-dom";
+import { useToast } from "../../custom-hooks/useToast";
+import { useAuth } from "../../context/authContext/authenticationContext";
+import { addVideoToHistoryOfUserService } from "../../services";
+import { useServices } from "../../context/servicesContext/servicesContext";
```
Comment (harshalini) [inline src/components/VideoCard/VideoCard.jsx:6]: Same here

File: src/components/modals/Submenu-modal/Submenu-modal.css [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  align-items: center;
+  position: absolute;
+  inset: 0;
+  background: grey;
+}
+
+.submenu-modal {
+  position: relative;
+  max-width: 300px;
+  min-height: 200px;
```
Comment (harshalini) [inline src/components/modals/Submenu-modal/Submenu-modal.css:13]: Just a suggestion, you can use rem for larger values

File: src/pages/VideoPage/VideoPage.jsx [hunk 4]
```diff
@@ -9,10 +9,11 @@ import { useData } from "../../context/dataContext/dataContext";
...
-  const { video } = useData();
+  const { videos } = useData();
   const { videoId } = useParams();
-  const currentVideo = video.find((item) => item._id === videoId);
-  const currentVideoCategory = video.filter(
+
+  const currentVideo = videos.find((item) => item._id === videoId);
+  const currentVideoCategory = videos.filter(
     (video) => video.category === currentVideo.category
   );
```
Comment (harshalini) [inline src/pages/VideoPage/VideoPage.jsx:18]: Nice use of inbuilt functions

File: src/reducers/dataReducer.js [hunk 5]
```diff
@@ -1,7 +1,9 @@
-const dataReducer = (state, action) => {
-  switch (action.type) {
+const dataReducer = (state, { type, payload }) => {
```
Comment (harshalini) [inline src/reducers/dataReducer.js:1]: I like that you have destructured here itself

**Review by harshalini — APPROVED** [review]
Code is really well written, liked the use of loader, functionalities are working fine!

**Pull request merged**